radsat-sk/operation/subsystems/RAntenna.c

```diff
@@ -0,0 +1,293 @@
+ISISantsI2Caddress RAntennaI2Caddress = {ANTENNA_I2C_SLAVE_ADDR_PRIMARY,ANTENNA_I2C_SLAVE_ADDR_REDUNANT};
+
+/** Struct that holds the current deployment status of the antenna*/
+antenna_deployment_status_t RAntennaStatus;
+
+/** Track whether the antenna has been initialized yet */
+static int antennaInitialized = 0;
+
+/** Track Number of times the antenna attempted to deploy*/
+static int antennaDeploymentAttempts = 0;
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Initializes the ISISpace Antenna driver
+ *
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaInit(void) {
+
+	// Only allow initialization once (return without error if already initialized)
+	if (antennaInitialized)
+		return 0;
+
+	// Isis Function call for antenna initialization
+	int error = IsisAntS_initialize(&RAntennaI2Caddress, ANTENNAS_ON_BOARD);
+
+	// Update flag if successfully initialized
+	if (!error)
+		antennaInitialized = 1;
+
+	// TODO: record errors (if present) to System Manager
+	return error;
+}
+
+/**
+ * Activate the auto-deployment mechanism on ISISpace Antenna System which will attempt to sequentially
+ * deploy all antennas present on the system without intervention. Has a built in timer per antenna deployment
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaDeploymentAttempt(void) {
+
+	// For Error detection
+	int error = 0;
+
+	// Initialize the union for antenna status
+	ISISantsStatus RISISantsStatus = { .fields = { 0 } };
+
+	// A Side deployment Attempt
+	while ( antennaDeploymentAttempts < MAX_DEPLOYMENT_ATTEMPTS ) {
+
+		// Get status of the antenna
+		error = IsisAntS_getStatusData(ANTENNA_I2C_SLAVE_ADDR_PRIMARY,isisants_sideA,&RISISantsStatus);
+
+		// Error check for requesting antenna status
+		if(error != 0) {
+
+			// TODO: record errors (if present) to System Manager
+			return error;
+		}
+
+		// Update the antenna status struct
+		RAntennaStatus.DeployedAntennaOne = !RISISantsStatus.fields.ant1Undeployed;
+		RAntennaStatus.DeployedAntennaTwo = !RISISantsStatus.fields.ant2Undeployed;
+		RAntennaStatus.DeployedAntennaThree = !RISISantsStatus.fields.ant3Undeployed;
+		RAntennaStatus.DeployedAntennaFour = !RISISantsStatus.fields.ant4Undeployed;
+		RAntennaStatus.AntennaArmed = RISISantsStatus.fields.armed;
+
+
+		// If one antenna is not deployed repeat attempt, otherwise do nothing
+		if (!RAntennaStatus.DeployedAntennaOne || !RAntennaStatus.DeployedAntennaTwo || !RAntennaStatus.DeployedAntennaThree || !RAntennaStatus.DeployedAntennaFour) {
+
+			// Arm A Side Antenna system
+			error = IsisAntS_setArmStatus(ANTENNA_I2C_SLAVE_ADDR_PRIMARY,isisants_sideA,isisants_arm);
```

Comment:
It would be better to specify the specific error. Eg. armingErroor, statusDataError, autoDeploymentError, so that it avoids name conflicts and makes the code more readable

Reply:
The only error that the IsisAntS_setArmStatus returns is from the hal/error.h. These warnings, although useful, don't outline the specific error that this case deals with. The approach of the struct check makes it so that in the specific case that one of the antennas doesn't deploy, attempt deployment again. I am of the opinion that this is quite readable and makes sense. Something to discuss during our meeting 

It should be noted that there is a "switch" of sort is on the antenna and it flipped when a successful deployment occurs. This means when there is another deployment attempt (as mentioned above) the antenna would skip any deployed antenna, not wasting time or computational power.